lib/jarvis/thread_logger.rb

```diff
@@ -0,0 +1,42 @@
+require "cabin"
+
+module Jarvis module ThreadLogger
+  class State
+    def initialize
+      @state = {}
+      @lock = Mutex.new
+    end
+
+    def <<(hash)
+      @lock.synchronize do
+        @state[Thread.current] = hash
```

Comment:
My idea here was to abuse the fact that threads are reused. This means that `status` will only ever report on the active thread pool members.

Here's an example:
```
Lita > lita slowcmd
Lita > lita slowcmd
Lita > lita slowcmd
Lita > lita slowcmd
Lita > lita slowcmd
Lita > lita slowcmd
Lita > lita slowcmd
Shell User: The NORMAL pool is full, so I am too busy to do that work right now, sorry! (shrug)
Lita > lita status
0: {:timestamp=>"2015-11-23T17:33:08.141708-0800", :message=>"Step 4"}
1: {:timestamp=>"2015-11-23T17:33:08.441435-0800", :message=>"Step 4"}
2: {:timestamp=>"2015-11-23T17:33:07.838728-0800", :message=>"Step 3"}
3: {:timestamp=>"2015-11-23T17:33:08.190267-0800", :message=>"Step 3"}
4: {:timestamp=>"2015-11-23T17:33:08.498478-0800", :message=>"Step 3"}
5: {:timestamp=>"2015-11-23T17:33:08.557320-0800", :message=>"Running command", :request=>"status"}
```